examples/cpp/custom_formula.cpp

```diff
@@ -0,0 +1,181 @@
+#include <cstdlib>
+#include <dlfcn.h>
+#include <cstdio>
+#include <fcntl.h>
+#include <unistd.h>
+#include <new>
+#include <memory>
+#include <algorithm>
+#include <chrono>
+
+#include "pf.h"
+
+#include "model/colormap.h"
+#include "model/image.h"
+#include "model/enums.h"
+#include "model/imagewriter.h"
+
+#include "model/fractfunc.h"
+#include "model/vectors.h"
+
+#define MAX_ITERATIONS 100
+
+constexpr double pos_params[N_PARAMS] {
+    0.0, 0.0, 0.0, 0.0, // X Y Z W
+    4.0, // Size or zoom
+    0.0, 0.0, 0.0, 0.0, 0.0, 0.0 // XY XZ XW YZ YW ZW planes (4D stuff)
+};
+
+int main() {
+    auto t_start = std::chrono::high_resolution_clock::now();
+
+    // initial setup: load fract4_stdlib globally so the loaded formula has access to it
+    void *fract_stdlib_handle = dlopen("./fract_stdlib.so", RTLD_GLOBAL | RTLD_NOW);
+    if (!fract_stdlib_handle) {
+        fprintf(stderr, "Error loading libfract_stdlib: %s", dlerror());
+        return -1;
+    }
+
+    // load formula lib
+    void *lib_handle = dlopen("./formula.so", RTLD_NOW);
+    if (!lib_handle)
+    {
+        fprintf(stderr, "Error loading formula: %s", dlerror());
+        return -1;
+    }
+    pf_obj *(*pfn)(void);
+    pfn = (pf_obj * (*)(void)) dlsym(lib_handle, "pf_new");
```

Comment:
This wants to be a reinterpret_cast<>() - one of the few non-UB uses of this cast type. See point 8 of the explantation section of https://en.cppreference.com/w/cpp/language/reinterpret_cast

Reply:
There's a lot of C-style stuff like this casting in the fract4dc module. As far as I know, in this case, C-style cast will end up behaving like reinterpret_cast, but I fully agree with you reinterpret_cast should be used explicitly for the sake of clarity.

Reply:
Unfortunately, the C-style cast actually invokes UB and issues a warning as it does a bit more than just a reinterpret_cast<>() - but yeah, this is something to look into and resolve.

I wouldn't want to put the C and C++ parts through a linter at the moment as it would be an overwhelming lot of output..